invenio_rdm_records/resources/serializers/ui/schema.py

```diff
@@ -94,6 +95,22 @@ def record_version(obj):
     return field_data
 
 
+def mask_removed_by(obj):
+    """Mask information about who removed the record."""
+    return_value = "Unknown"
+    removed_by = obj.get("removed_by", None)
+
+    if removed_by is not None:
+        user = removed_by.get("user", None)
+
+        if user == "system":
+            return_value = "System (automatic)"
+        elif user is not None:
+            return_value = "User"
+
+    return _(return_value)
```

Comment:
I am afraid this won't work on extract messages step, since it is runtime evaluation, to be tested